addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_animations.py

```diff
@@ -109,21 +109,26 @@ def __gather_animation(blender_action: bpy.types.Action,
     if not __filter_animation(blender_action, blender_object, export_settings):
         return None
 
-    name = __gather_name(blender_action, blender_object, export_settings)
-    try:
-        animation = gltf2_io.Animation(
-            channels=__gather_channels(blender_action, blender_object, export_settings),
-            extensions=__gather_extensions(blender_action, blender_object, export_settings),
-            extras=__gather_extras(blender_action, blender_object, export_settings),
-            name=name,
-            samplers=__gather_samplers(blender_action, blender_object, export_settings)
-        )
-    except RuntimeError as error:
-        print_console("WARNING", "Animation '{}' could not be exported. Cause: {}".format(name, error))
-        return None
-
-    if not animation.channels:
-        return None
+    animation = None
+
+    export_user_extensions('pre_gather_animation_hook', export_settings, animation, blender_action, blender_object)
+
+    if animation is None:
```

Comment:
Not sure this last change is good.
If the object is TRS animated and you add a custom animation, TRS will not be exported

Reply:
@pepperoni505 ?

Reply:
Hi @julienduroure. I was thinking of different ways to do this - essentially the issue is the addon I'm writing doesn't use the `channels` part of the animation, so this will always return `None` for those. I see what you mean about the TRS issue - I can undo that commit, but we still have the issue of checking if `channels` are present

Reply:
I've figured out how to fix this on my end and have undone that commit